js/piwik.js

```diff
@@ -1393,6 +1393,14 @@ if (typeof window.Piwik !== 'object') {
             var now;
 
             executePluginMethod('unload');
+
+            now  = new Date();
+            var aliasTime = now.getTimeAlias();
+            if( (expireDateTime - aliasTime) > 3000) // fix bug #12108
+            {
+                return;
```

Comment:
Instead of `return;` can we maybe better set `expireDateTime = aliasTime + 3000` so we still wait up to 3 seconds when this happens? Otherwise we might ignore some tracking requests on unload.

Reply:
Hi @yossicohn could you review @tsteur comment and apply feedback to the PR?

Reply:
```suggestion
                expireDateTime = aliasTime + 3000
```